create/application.go

```diff
@@ -102,10 +103,21 @@ func (app *applicationCmd) Run(ctx context.Context, client *api.Client) error {
 		if err := auth.Valid(); err != nil {
 			return fmt.Errorf("the credentials are given but they are empty: %w", err)
 		}
-		// for git auth we create a separate secret and then reference it in the app.
+
 		secret := auth.Secret(newApp)
+		// for git auth we create a separate secret and then reference it in the app.
 		if err := client.Create(ctx, secret); err != nil {
-			return fmt.Errorf("unable to create git auth secret: %w", err)
+			if k8sError.IsAlreadyExists(err) {
+				// only update the secret if it is managed by nctl in the first place
+				if _, exists := newApp.Annotations[util.ManagedByAnnotation]; exists {
```

Comment:
this only checks if the managed by annotation key is there but the value is also important :) it should also check if the value is `util.NctlName`.

```suggestion
				if v, exists := newApp.Annotations[util.ManagedByAnnotation]; exists && v == util.NctlName {
```